### DISCUSSION TITLE
홍동화

### DISCUSSION THREAD

**Discussion by z1nun** [z1nun/ASTERISK-#21] [독서일지]
# 각자 선정한 개발자 필독서를 읽고 그날 읽은 내용을 간단히 정리해서 적어주세요!

## REFACTORING 2판

- 읽은 내용은 밑에 comments에 적어주세요!

**Comment by fairytale779** [comment]
22.10.02 - 03 (코드가 너무 많아서 이틀로 나눠 읽음..)
일단 리팩토링의 목적은, 코드를 볼 때 더욱 편하고 수정을 간편하게 하기 위해 하는 작업이다. 중복제거, 프로그램 작성 속도 개선, 소프트웨어 속도 향상 등을 위하며, 기능 추가 버그 수정 등이 기능이 있다. 긴 코드를 수정할 때면 먼저 프로그램의 작동 방식을 더 쉽게 파악할 수 있게 코드를 여러 함수와 여러 프로그램을 재구성하는 것이 편리하다.. 구조를 먼저 잡은 후에 기능을 수정하는 것이(기능을 추가하기 쉬운 형태로 변경하는 것) 작업이 훨씬 수월하다. 
그리고 수시로 테스트를 한다. 나중에 한꺼번에 작업하려면 오래 걸리고 찾는데에 더 많은 시간을 쏟는다.
하나의 리팩터링을 문제 없이 끝낼 때마다 커밋하며, 자잘한 병경들이 어느정도 의미있는 단위로 뭉쳐지면 푸시한다.

**Reply by fairytale779** [reply]
22.10.04 첫번째 예시 코드 따라 치며 공부.
청구내역을 HTML로 출력하는 기능이 필요.
예측 가능한 기획 변경 대응
비슷한 함수를 복사할 경우 중복되는 부분의 수정이 일관되게 반영되도록 보장해야한다.
리팩터링이 필요한 이유는 변경때문이다.

**Math.max()**함수는 입력값으로 받은 0개 이상의 숫자 중 가장 큰 숫자를 반환합니다.
Math.floor() 함수는 주어진 숫자와 같거나 작은 정수 중에서 가장 큰 수를 반환합니다.
